src/functional/histogram_compression.zig

```diff
@@ -0,0 +1,519 @@
+// Copyright 2024 TerseTS Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of the Min-Merge algorithm from the paper
+//! "Buragohain, Chiranjeeb, Nisheeth Shrivastava, and Subhash Suri.
+//! Space efficient streaming algorithms for the maximum error histogram.
+//! IEEE ICDE 2006.
+//! https://doi.org/10.1109/ICDE.2007.368961.
+//! Min-Merge find a Piecewise Constant Histogram compressed representation
+//! of the time series. Thus, we called the method PWCH as in the benchmark paper
```

Comment:
Do we really need to do so? I think just referencing the paper is enough. I remove the word benchmark to avoid confusion. Now it looks like this:
//! Thus, the abbreviation PWCH is used for the method as in the paper:
//! https://doi.org/10.1109/TKDE.2012.237.